packages/RichTextEditing-Core.package/RichTextDocument.class/instance/wordcount.st

```diff
@@ -0,0 +1,16 @@
+count
+wordcount
+	
+	| byteString count i readWord|
+	count := 0.
+	readWord := false.
+	byteString := self text string asByteArray.
+	
+	i := 1.
+	[i <= byteString size] whileTrue:[
+		(readWord and: (self emptyChars includes: (byteString at: i))) ifTrue: [readWord := false. count := count+1].
+		(self emptyChars includes: (byteString at: i)) ifFalse: [readWord := true].
+		i := i + 1
+		].
+	readWord ifTrue: [count := count +1].
```

Comment:
just tested this small code snippet which checks if the previous element is an emptyChar and the current not:
```
| prev bool |
prev := 32.
^ (#[104 97 108 32 108 32 32 111] count: [:each | bool := ((#[32] includes: each) not and: (#[32] includes: prev)). prev := each. bool ]) 
```
Not sure if better, however more smalltalk like

Reply:
May not detect words appropriately when words are seperated by multiple seperator chars

Reply:
If both previous and current are an empty charstring the bool returns false and is not counted. I think every word is an empty charstring followed by a not empty charstring.

Reply:
implemented with 
350982f